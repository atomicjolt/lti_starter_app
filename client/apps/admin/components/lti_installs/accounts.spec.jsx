import React from 'react';
import TestRenderer from 'react-test-renderer';
import Accounts from './accounts';

describe('lti installs accounts', () => {
  let result;

  beforeEach(() => {
    const props = {
      accounts: {
        1: {
          id: 1,
          name: 'account_name',
          parent_account_id: null
        }
      },
      setAccountActive: () => {},
      currentAccount: null,
    };
    result = TestRenderer.create(<Accounts {...props} />);
  });

  it('renders', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
