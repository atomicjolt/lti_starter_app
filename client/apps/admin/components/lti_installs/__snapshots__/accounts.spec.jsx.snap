// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti installs accounts matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Accounts
    accounts={
      Object {
        "1": Object {
          "id": 1,
          "name": "account_name",
          "parent_account_id": null,
        },
      }
    }
    currentAccount={null}
    setAccountActive={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SubAccount
          account={
            Object {
              "id": 1,
              "name": "account_name",
              "parent_account_id": null,
            }
          }
          accounts={
            Object {
              "1": Object {
                "id": 1,
                "name": "account_name",
                "parent_account_id": null,
              },
            }
          }
          currentAccount={null}
          isActive={null}
          setAccountActive={[Function]}
        />,
      ],
      "className": "c-filter-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "account_1",
        "nodeType": "class",
        "props": Object {
          "account": Object {
            "id": 1,
            "name": "account_name",
            "parent_account_id": null,
          },
          "accounts": Object {
            "1": Object {
              "id": 1,
              "name": "account_name",
              "parent_account_id": null,
            },
          },
          "currentAccount": null,
          "isActive": null,
          "setAccountActive": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SubAccount
            account={
              Object {
                "id": 1,
                "name": "account_name",
                "parent_account_id": null,
              }
            }
            accounts={
              Object {
                "1": Object {
                  "id": 1,
                  "name": "account_name",
                  "parent_account_id": null,
                },
              }
            }
            currentAccount={null}
            isActive={null}
            setAccountActive={[Function]}
          />,
        ],
        "className": "c-filter-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "account_1",
          "nodeType": "class",
          "props": Object {
            "account": Object {
              "id": 1,
              "name": "account_name",
              "parent_account_id": null,
            },
            "accounts": Object {
              "1": Object {
                "id": 1,
                "name": "account_name",
                "parent_account_id": null,
              },
            },
            "currentAccount": null,
            "isActive": null,
            "setAccountActive": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
